drivers/interrupt_controller/intc_nxp_irqsteer.c

```diff
@@ -225,20 +225,36 @@
 #include <zephyr/device.h>
 #include <zephyr/devicetree/interrupt_controller.h>
 #include <zephyr/irq.h>
-#include <fsl_irqsteer.h>
 #include <zephyr/cache.h>
 #include <zephyr/sw_isr_table.h>
 #include <zephyr/logging/log.h>
 #include <zephyr/pm/device_runtime.h>
 #include <zephyr/pm/device.h>
-
+#include <zephyr/sys/util.h>
 #include "sw_isr_common.h"
 
-LOG_MODULE_REGISTER(nxp_irqstr);
+LOG_MODULE_REGISTER(nxp_irqsteer);
 
-/* used for driver binding */
+/* Driver compatibility */
 #define DT_DRV_COMPAT nxp_irqsteer_intc
 
+/* IRQ_STEER register offsets */
+#define CTRL_STRIDE_OFF(_t, _r)	(_t * 4 * _r)
+#define CHAN_MASK(n, t)		(CTRL_STRIDE_OFF(t, 0) + 0x4 * (n) + 0x4)
+#define CHAN_SET(n, t)		(CTRL_STRIDE_OFF(t, 1) + 0x4 * (n) + 0x4)
+#define CHAN_STATUS(n, t)	(CTRL_STRIDE_OFF(t, 2) + 0x4 * (n) + 0x4)
+#define CHAN_MINTDIS(t)		(CTRL_STRIDE_OFF(t, 3) + 0x4)
+#define CHAN_MASTRSTAT(t)	(CTRL_STRIDE_OFF(t, 3) + 0x8)
+
+/* IRQSTEER register bits width is 32 */
+#define IRQSTEER_INT_SRC_REG_WIDTH 32
+/* IRQSTEER AGGREGATED(fixed) interrupt number per group is 64 */
+#define IRQSTEER_AGGRAGATED_INT_NUM_PER_GRP 64
+
+/* Register access macros */
+#define IRQSTEER_READ32(base, offset)       sys_read32((base) + (offset))
+#define IRQSTEER_WRITE32(base, offset, val) sys_write32((val), (base) + (offset))
+
```

Comment:
I would get read read of this macros because capital letter make code harder to read.

Reply:
> I would get read read of this macros because capital letter make code harder to read.

Sure, np. I will update it(irqsteer_read_reg32).